File: tests/compile-fail/ForeignNullableGenericImplementedSlot.kt
// #86 — the crossing at the IMPLEMENTING position, on a PLAIN .NET interface.
//
// A call is not the only way to meet an uninhabitable slot. A Kotlin class can DERIVE from a .NET type that declares
// one, and there the crossing is in the slot the class must FILL rather than in anything it calls — so the call-side
// refusal never saw it and the class died at load. Emitting the declaration's own signature would not fix it: the
// Kotlin body still reads the argument as Kotlin's `List<object>`, which would move the mismatch out of load time
// and into the body. No valid CIL lowering exists, so the author is owed the message.
import plainnet.ITake
import System.Collections.Generic.List

class CI : ITake {
    override fun Take(xs: List<Int?>): String = "I:ok"
}

fun main() {
    println(CI().toString().substring(0, 2))
}

File: tests/compile-fail/ForeignNullableGenericImplementedSlot.cs
using System;
using System.Collections.Generic;

// A PLAIN .NET surface — no DotKt metadata of any kind. That provenance is the point: the carrier machinery that
// repairs an erased slot reads DotKt attributes, so a BCL or third-party interface has nothing for it to read, and
// this whole column fell through to a load-time failure with our type's name on it.
namespace plainnet
{
    // An uninhabitable slot on an INTERFACE. `class C : ITake` used to compile clean and die with
    // "Signature of the body and declaration in a method implementation do not match".
    public interface ITake { string Take(List<int?> xs); }

    // The ABSTRACT BASE twin, which died with "does not have an implementation" instead.
    public abstract class BTake { public abstract string Take(List<int?> xs); }
}
